src/subprocess-posix.cc

```diff
@@ -333,13 +333,13 @@ bool SubprocessSet::DoWork() {
     int fd = (*i)->fd_;
     if (fd >= 0 && FD_ISSET(fd, &set)) {
       (*i)->OnPipeReady();
```

Comment:
this is existing already, changes below have nothing to do with the fd value...

if an infinite loop is still possible after the fd value is handled then `continue` should cause increment of `i`, like how a normal for loop is written...

are we sure that omitting the increment in the loop itself isn't the actual mistake and all you need to do is put it back?